### PR TITLE
feat(instance): data source image

### PR DESCRIPTION
**Pull request by kindermoumoute**

### CODE REVIEW

File: scaleway/data_source_instance_image.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+			"name": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Description:   "exact name of the desired image",
+				ConflictsWith: []string{"image_id"},
+			},
+			"image_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Description:   "exact name of the desired image",
```
Comment (jerome-quere) [inline scaleway/data_source_instance_image.go]: wrong description

File: scaleway/data_source_instance_image.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+			},
+			"zone":            zoneSchema(),
+			"organization_id": organizationIDSchema(),
+
+			"public": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "indication if the image is public",
+			},
+			"bootscript_id": {
```
Comment (jerome-quere) [inline scaleway/data_source_instance_image.go]: `default_bootscript_id`

File: scaleway/data_source_instance_image.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+		return err
+	}
+
+	imageID, ok := d.GetOk("image_id")
+	if !ok {
+		res, err := instanceApi.ListImages(&instance.ListImagesRequest{
+			Zone: zone,
+			Name: expandStringPtr(d.Get("name")),
+			Arch: expandStringPtr(d.Get("architecture")),
+		})
```
Comment (jerome-quere) [inline scaleway/data_source_instance_image.go]: add with all pages  option

File: scaleway/data_source_instance_image.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+		if err != nil {
+			return err
+		}
+		if len(res.Images) == 0 {
+			return fmt.Errorf("no image found with the name %s and architecture %s", d.Get("name"), d.Get("architecture"))
+		}
+		if len(res.Images) > 1 {
+			return fmt.Errorf("%d images found with the same name %s and architecture %s", len(res.Images), d.Get("name"), d.Get("architecture"))
```
Comment (jerome-quere) [inline scaleway/data_source_instance_image.go]: Waht about the case we have 2 image names `ubuntu-bionic` and `ubuntu-bionic-ml` this condition will not work

File: scaleway/data_source_instance_image.go [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+		}
+		imageID = res.Images[0].ID
+	}
+
+	resp, err := instanceApi.GetImage(&instance.GetImageRequest{
+		Zone:    zone,
```
Comment (jerome-quere) [inline scaleway/data_source_instance_image.go]: Zone may be contain inside the image_id attribute.

File: website/docs/d/instance_image.html.markdown [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+# scaleway_instance_image
+
+Gets information about an instance image.
+
+## Example Usage
+
+```hcl
+// Get info by image name
+data "scaleway_instance_image" "my_key" {
```
Comment (QuentinBrosse) [inline website/docs/d/instance_image.html.markdown]: `my_key` is an image

File: website/docs/d/instance_image.html.markdown [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  name  = "ubuntu bionic"
+}
+
+// Get info by image id
+data "scaleway_instance_image" "my_key" {
```
Comment (QuentinBrosse) [inline website/docs/d/instance_image.html.markdown]: `my_key` is an image

File: website/docs/d/instance_image.html.markdown [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+}
+```
+
+## Argument Reference
+
+- `name` - (Optional) The image name. Only one of `name` and `image_id` should be specified.
+
+- `image_id` - (Optional) The image id. Only one of `name` and `image_id` should be specified.
+
+- `architecture` - (Optional, default `x86_64`) The architecture the image is compatible with.Possible values are: `x86_64` or `arm`.
```
Comment (QuentinBrosse) [inline website/docs/d/instance_image.html.markdown]: with. Possible

File: website/scaleway.erb [hunk 9]
```diff
@@ -178,6 +178,9 @@
             <li>
               <a href="/docs/providers/scaleway/d/security_group.html">scaleway_security_group</a>
             </li>
+            <li>
+              <a href="/docs/providers/scaleway/d/image.html">scaleway_image</a>
```
Comment (QuentinBrosse) [inline website/scaleway.erb]: You need to add a deprecation message to [`scaleway_image`](https://github.com/terraform-providers/terraform-provider-scaleway/blob/master/scaleway/data_source_image.go) schema.

File: scaleway/data_source_instance_image.go [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+func dataSourceScalewayInstanceImageRead(d *schema.ResourceData, m interface{}) error {
+	meta := m.(*Meta)
+	instanceApi, zone, err := getInstanceAPIWithZone(d, meta)
+	if err != nil {
+		return err
+	}
+
+	imageID, ok := d.GetOk("image_id")
+	if !ok {
+		res, err := instanceApi.ListImages(&instance.ListImagesRequest{
```
Comment (QuentinBrosse) [inline scaleway/data_source_instance_image.go]: Add a `// Get instance by name, zone, and arch.`

File: scaleway/data_source_instance_image.go [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+			Zone: zone,
+			Name: expandStringPtr(d.Get("name")),
+			Arch: expandStringPtr(d.Get("architecture")),
+		}, scw.WithAllPages())
+		if err != nil {
+			return err
+		}
+		if len(res.Images) == 0 {
+			return fmt.Errorf("no image found with the name %s and architecture %s", d.Get("name"), d.Get("architecture"))
```
Comment (QuentinBrosse) [inline scaleway/data_source_instance_image.go]: and `zone`

File: scaleway/data_source_instance_image.go [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+		}
+		if len(res.Images) > 1 && !d.Get("latest").(bool) {
+			return fmt.Errorf("%d images found with the same name %s and architecture %s", len(res.Images), d.Get("name"), d.Get("architecture"))
```
Comment (QuentinBrosse) [inline scaleway/data_source_instance_image.go]: and `zone`

File: website/docs/d/instance_image.html.markdown [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+# scaleway_instance_image
+
+Gets information about an instance image.
+
+## Example Usage
+
+```hcl
+// Get info by image name
+data "scaleway_instance_image" "my_image" {
+  name  = "ubuntu bionic"
```
Comment (jerome-quere) [inline website/docs/d/instance_image.html.markdown]: Don't feature a use-case like ubuntu. These use-case should be cover by `scaleway_marketplace_image`
Use a name like `my-image-name`

**Comment by jerome-quere** [comment]
close #276

**Pull request merged**